app/models/local_census_record.rb

```diff
@@ -1,8 +1,12 @@
 class LocalCensusRecord < ApplicationRecord
+  ALLOWED_DOCUMENT_TYPES = { dni: "1", passport: "2", residence_card: "3" }.freeze
+
   before_validation :sanitize
 
   validates :document_number, presence: true
   validates :document_type, presence: true
+  validates :document_type, inclusion: { in: ALLOWED_DOCUMENT_TYPES.values },
+                            if: -> { document_type.present? }
```

Comment:
Good question @javierm, what i was trying to do is to not validate inclussion when document_type is empty, but i think a better way to do the same should be to use allow_blank validation option to achieve the same.

`validates :document_type, presence: true, inclusion: { in: ALLOWED_DOCUMENT_TYPES.values, allow_blank: true }`

The inclussion validation is only needed when running import of local census records or if anyone is creating records from `rails console`, to help the user to understand what values are the accepted ones. This problems does not exists when an administrator creates local census records through the web interface cause we are forcing to select `document_type`from correct values always.

What do you think? Maybe you prefer to run both validation always?

Reply:
Finally update inclussion validation with allow_blank option set to true in place of using an if condition.

Reply:
I've just realized `document_type` cannot be empty because the line above this one says:

```
 validates :document_type, presence: true
```

So the `allow_blank` option doesn't seem necessary :thinking:. I've tried creating a record without a document type and I get a validation error because of the `presence` condition.

So what do you think about:

```
 validates :document_type, presence: true, inclusion: { in: ["1", "2", "3"] }
```

Reply:
I've just realized `document_type` cannot be empty because the line above this one says:

```
 validates :document_type, presence: true
```

So the `allow_blank` option doesn't seem necessary :thinking:. I've tried creating a record without a document type and I get a validation error because of the `presence` condition.

So what do you think about:

```
 validates :document_type, presence: true, inclusion: { in: ["1", "2", "3"] }
```

Reply:
Yes, of course, but there is a little difference between both versions. Without `allow_blank` option model will have 2 validation errors when the `document_type` is empty, the will see:

1. can't be blank
2 is not included in the list, allowed values are 1 for DNI, 2 for Passport and 3 for Residence Card

When the `allow_blank` option is present and set to `true` the user will see only one validation error at a time.

Reply:
Interesting! I didn't know that.